cmake/Helpers.cmake

```diff
@@ -0,0 +1,685 @@
+        # only later c++ standards conditionally allow function/object pointer casts
+        # gcc pragmas for disabling are broken: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53431
+        IF(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7)
+            STRING(APPEND CMAKE_CXX_FLAGS " -Wno-error=pedantic")
+        ENDIF ()
+
+        ADD_FLAG_IF_SUPPORTED(FLAG "-Wimplicit-fallthrough" VALUE "2")
+    ENDIF ()
+
+    IF(CMAKE_COMPILER_ID MATCHES "Clang")
+        IF(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6 OR APPLE)
+            STRING(APPEND CMAKE_CXX_FLAGS " -Wno-error=pedantic")
+        ENDIF ()
+    ENDIF ()
+
+    IF(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+        STRING(APPEND CMAKE_CXX_FLAGS " /fp:fast") 
+    ENDIF()
+
+    USE_SCCACHE_IF_AVAILABLE()
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_PYTHON)
+    IF(PYTHON_VERSION STREQUAL "Auto")
+        FIND_PACKAGE(PythonLibs 3)
+        # Record for reporting later
+        SET_INTERNAL(PYTHON_VERSION 3)
+    ELSEIF(PYTHON_VERSION STREQUAL "2")
+        MESSAGE(FATAL_ERROR "Python 2 support has been removed, rerun cmake in a clean build dir.")
+    ELSE()
+        FIND_PACKAGE(PythonLibs ${PYTHON_VERSION} REQUIRED)
+    ENDIF()
+    IF(PYTHONLIBS_FOUND)
+        MESSAGE(STATUS "Looking for Python libraries and headers: found")
+        INCLUDE_DIRECTORIES(SYSTEM ${PYTHON_INCLUDE_DIRS})
+        SET_INTERNAL(PYTHON_LIBRARIES "${PYTHON_LIBRARIES}")
+    ENDIF()
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_SDL SDL_BACKEND)
+    # Autodetection mechanism
+    # Default priority to SDL
+    # If both backends are found, preferring SDL2
+    IF(SDL_BACKEND STREQUAL "Auto")
+        MESSAGE(STATUS "Autodetecting SDL backend...")
+        IF(WIN32 AND _VCPKG_INSTALLED_DIR)
+            FIND_PACKAGE(SDL2 CONFIG REQUIRED)
+            GET_TARGET_PROPERTY(SDL2_INCLUDE_DIRS SDL2::SDL2 INTERFACE_INCLUDE_DIRECTORIES)
+            GET_TARGET_PROPERTY(SDL2_LIBRARY_DEBUG SDL2::SDL2 IMPORTED_IMPLIB_DEBUG)
+            GET_TARGET_PROPERTY(SDL2_LIBRARY_RELEASE SDL2::SDL2 IMPORTED_IMPLIB_RELEASE)
+            SET(SDL2_LIBRARIES debug ${SDL2_LIBRARY_DEBUG} optimized ${SDL2_LIBRARY_RELEASE})
+        ELSE()
+            FIND_PACKAGE(SDL2)
+        ENDIF()
+        IF(SDL2_FOUND)
+            MESSAGE(STATUS "Found SDL 2.0, switching to SDL2 backend.")
+            SET_INTERNAL(SDL_BACKEND_DETECTED "SDL2")
+            SET_INTERNAL(SDL_FOUND ${SDL2_FOUND})
+        ELSE()
+            INCLUDE(FindSDL)
+            IF(SDL_FOUND)
+                MESSAGE(STATUS "Found SDL 1.2, switching to SDL backend.")
+                SET_INTERNAL(SDL_BACKEND_DETECTED "SDL")
+                SET_INTERNAL(SDL_FOUND ${SDL_FOUND})
+            ENDIF()
+        ENDIF()
+        IF(NOT (SDL_FOUND OR SDL2_FOUND))
+            MESSAGE(WARNING "No SDL or SDL2 libraries found.")
+            MESSAGE(FATAL_ERROR "Please get SDL from www.libsdl.org")
+        ENDIF()
+    ENDIF()
+
+    IF(SDL_BACKEND STREQUAL "SDL" OR SDL_BACKEND_DETECTED STREQUAL "SDL")
```

Comment:
I don't know why you haven't answered me on this, but I see no good reason for these doubled variables.

Reply:
It is to keep `SDL_BACKEND` value equal to user's choice and `SDL_BACKEND_DETECTED` equal to "real" detected value. It is made specifically for case with `Auto` as a chosen value.

Reply:
Auto is just the default, there's no reason for anyone to pass it. If one wanted to change the backend used in a subsequent build, a reconfigure is needed either way, so there's nothing to gain.

And same with OpenGL, except there's not even an auto there.